Update docs (#73)
makinacorpus/DbToolsBundle: docs/content/backup_restore.md

# Backup and Restore

The *DbToolsBundle* comes with two Symfony Console commands to backup and restore
your database but also a tiny backups manager (`DbToolsStorage`) which handle backup
files for you.

## Backup command

> The backup command will use [defined binary]('./configuration#binaries') for your database vendor with correct parameters

Reviewer: Pardon my french.

> The backup command will use [defined binary]('./configuration#binaries') for your database vendor with correct parameters

Author: configured ?

> The backup command will use [defined binary]('./configuration#binaries') for your database vendor with correct parameters

Reviewer: I'm not sure about what you are trying to say, are they "predefined" as you pre-configured them, are they "configured" as the user did configure it, or they just are because it's hardcoded ?

> The backup command will use [defined binary]('./configuration#binaries') for your database vendor with correct parameters

Author: I updated with `predefined or configured binary`

to dump your database.

```sh
console db-tools:backup
```

> You can specify the behaviour of the command with some options:

Reviewer 2: ```suggestion
You can specify the behaviour of the command with some options detailed below.
```

### Connection

Per default, the command will backup the database from the default DBAL connection.

You can choose to backup a database from another connection with `--connection` option:

```sh
console db-tools:backup --connection other_connection_name
```

### Excluded tables

You may have configured [tables to be excluded in the bundle configuration](./configuration#excluded-tables).
If so, these tables will be automatically excluded each time you launch the command.

But if you want to temporary exclude some tables, run the command with the `--excluded-table` option:

```sh
# Exclude a table
console db-tools:backup --excluded-table table_to_exclude

# Or more
console db-tools:backup --excluded-table table_to_exclude_1 --excluded-table table_to_exclude_2
```

### No cleanup

> Each time you launch the backup command, [backup file is store in a directory](./configuration#storage-directory).

Reviewer: I probably would have documented the cleanup and its configuration before documenting how to disable it.

> Each time you launch the backup command, [backup file is store in a directory](./configuration#storage-directory).

Reviewer 2: ```suggestion
Each time you launch the backup command, [a backup file is stored in a directory](./configuration#storage-directory).
```

With time, this directory will grow, that's why a [backup expiration age](./configuration#storage-directory#backup-expiration-age)
was added. Every time you launch the command, at the end, it will be asked if you want to remove obsolete

> backup files (ie files that have pasted their expiration date).

Reviewer 2: ```suggestion
backup files (i.e. files that have passed their expiration date).
```

If you want to skip this step, launch it with option `--no-cleanup':

```sh
console db-tools:backup --no-cleanup
```

:::warning

> Note that using this option, backup files will never be cleaned up

Reviewer 2: ```suggestion
Note that by using this option, backup files will never be cleaned up.
```

Author: I'm not sure about adding this 'by'

:::

## Restore command

The restore command will use [defined binary]('./configuration#binaries') for your database vendor with correct parameters

> to restore your database from existing backup files.

Reviewer 2: ```suggestion
to restore your database from an existing backup file.
```

```sh
console db-tools:restore
```

> You can specify the behaviour of the command with some options:

Reviewer: It weird have a sentence terminated by `:` followed by a title ?

### Connection

Per default, the command will restore the database from the default DBAL connection.

You can choose to restore a database from another connection with `--connection` option:

```sh
console db-tools:restore --connection other_connection_name
```

### Filename

When you launch this command, existing backup files will be listed and it
will be asked you to choose one to restore.

If you want to skip this step, or if your backup file is unknown to the DbToolsStorage,
you can specify a file to restore with the `--filename` option:

```sh
console db-tools:restore --filename /path/to/my/backup.sql
```

### Force

> Each time you ru this command, as it is not a sensitive operation, a confirmation will

Reviewer: I love to RU things myself.

> Each time you ru this command, as it is not a sensitive operation, a confirmation will

Reviewer 2: ```suggestion
Each time you run this command, as it is a sensitive operation, a confirmation will
```

```sh
console db-tools:backup --no-cleanup
```

:::warning
Note that using this option, backup files will never be cleaned up
:::

## Restore command

The restore command will use [predefined or configured binary](./configuration#binaries) for your database vendor with correct parameters
to restore your database from existing backup files.

```sh
console db-tools:restore
```

You can specify the behaviour of the command with some options.

### Connection

By default, the command will restore the database from the default DBAL connection.

You can choose to restore a database from another connection with `--connection` option:

```sh
console db-tools:restore --connection other_connection_name
```

### Filename

When you launch this command, existing backup files will be listed and it
will be asked you to choose one to restore.

If you want to skip this step, or if your backup file is unknown to the DbToolsStorage,
you can specify a file to restore with the `--filename` option:

```sh
console db-tools:restore --filename /path/to/my/backup.sql
```

### Force

Each time you ru this command, as it is not a sensitive operation, a confirmation will

> be asked. If you want to skip it, use the `--force` option.

Reviewer 2: I wonder if it's not the role of the default `--no-interaction` Symfony console option?

Reviewer 2: I should have read a bit more before commenting. There's a security for the production environment. That's not the case with `--no-interaction`.

```sh
console db-tools:restore --force
```

:::danger
By default, this command does not allow to restore a database when used in **production environment**.
Even if you use it with `--force`.

If you know what you are doing and want to restore a
backup in production, use the `--yes-i-am-sure-of-what-i-am-doing` option.

Note that, even with this option, the command will ask you to confirm, *twice*, that you

> really want to do so.

Reviewer 2: Is `--no-interaction` option supported for the needs of automated deployment process?

Author: nop'

:::

## Storage

> As mentioned earlier on this page, the *DbToolsBundle* stores can list existing backup files

Reviewer 2: ```suggestion
As mentioned earlier on this page, the *DbToolsBundle* can list existing backup files
```

when you want to restore a previous one with the restore command.

All backups are stored in a directory. By default this directory is `%kernel.project_dir%/var/db_tools`

> but [it could be configured](./configuration#storage-directory).

Reviewer 2: ```suggestion
but [it could be customized](./configuration#storage-directory).
```

Author: you reviewed an outdated version :grimacing:
